src/Azure.Functions.Cli/Common/Utilities.cs

```diff
@@ -160,7 +160,14 @@ internal static async Task<T> SafeExecution<T>(Func<Task<T>> action)
 
         internal static string EnsureCoreToolsLocalData()
         {
-            var localPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "azure-functions-core-tools");
+            var appDataDir = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+
+            if (string.IsNullOrEmpty(appDataDir))
+            {
+                throw new Exception("Cannot find the Local Application Data.");
```

Comment:
throwing an exception here makes sense as this is an error condition. In case of telemetry, this will be swallowed by the caller.